app/controllers/admin/users_controller.rb

```diff
@@ -103,7 +103,12 @@ def user_status
       rescue StandardError => e
         flash[:danger] = e.message
       end
-      redirect_to admin_user_path(params[:id])
+
+      if request.referer&.include?(params[:id])
```

Comment:
I think this is an "arbitrary" situation, but let's say the referrer URL was `hello.com/123` and the `params[:id]` was 12, we'd have a match.

Does the referrer have a preceding and trailing slash? `.include?("/#{params[:id]}/")`;  this isn't quite a blocker, but wanted to raise attention to a possible small bug.

Reply:
I originally had it as `request.referer&.include?(admin_user_path(params[:id]))` but thought that might be overkill. While the referrer does have a proceeding slash, adding it to the `include?` query like you suggested doesn't feel any less buggy than checking if it includes just `params[:id]`, since the Admin still could come from another page that contains `params[:id]` with the slash and still fall within the conditional. 

This all being said, I'll change it back to my more overkill way of doing it, @jeremyf